backend/qemu.pm

```diff
@@ -540,22 +546,16 @@ sub start_qemu {
     $vars->{BIOS} //= $vars->{UEFI_BIOS} if ($vars->{UEFI});    # XXX: compat with old deployment
     $vars->{UEFI} = 1 if $vars->{UEFI_PFLASH};
 
-    if ($vars->{UEFI_PFLASH} && $vars->{ARCH} eq 'x86_64' && !$vars->{BIOS}) {
-        foreach my $firmware (@bmwqemu::ovmf_locations) {
-            if (-e $firmware) {
-                $vars->{BIOS} = $firmware;
-                last;
-            }
-        }
-        if (!$vars->{BIOS}) {
-            # We know this won't go well.
-            die "No UEFI firmware can be found! Please specify BIOS or UEFI_BIOS or install an appropriate package";
-        }
+    if ($vars->{UEFI_PFLASH}) {
+        $vars->{BIOS} //= find_ovmf =~ s/-code//r;
+    }
+    elsif ($vars->{UEFI}) {
+        $vars->{UEFI_PFLASH_CODE} //= find_ovmf;
+        $vars->{UEFI_PFLASH_VARS} //= $vars->{UEFI_PFLASH_CODE} =~ s/code/vars/r;
+        die "No UEFI firmware can be found! Please specify UEFI_PFLASH_CODE/UEFI_PFLASH_VARS or BIOS or UEFI_BIOS or install an appropriate package" unless $vars->{UEFI_PFLASH_CODE};
```

Comment:
"or install an appropriate package" is kinda vague. i think it can be confusing. Can you elaborate a bit?

Reply:
it's the same message as in before, I have not really changed the content. As package names differ a lot among different distributions I consider it not feasible to suggest a specific package.